docs/ref/models/querysets.txt

```diff
@@ -1120,6 +1120,35 @@ Sometimes, the Django query syntax by itself can't easily express a complex
 ``QuerySet`` modifier — a hook for injecting specific clauses into the SQL
 generated by a ``QuerySet``.
 
+.. admonition:: Use this method as a last resort
+
+    This is an old API that we aim to deprecate at some point in the future.
+    Use it only if you cannot express your query using other queryset methods.
+    If you do need to use it, please `file a ticket
+    <https://code.djangoproject.com/>`_ with your use case so that we can
```

Comment:
What about making sure all tickets about `extra()` use the tag you've started assigning and also suggest having a look at them before filing a new ticket?